src/Tasks/AssemblyDependency/ResolveAssemblyReference.cs

```diff
@@ -2206,36 +2206,48 @@ out autoUnifiedRemappedAssemblyReferences
                         );
                     }
 
-                    IReadOnlyCollection<DependentAssembly> allRemappedAssemblies = CombineRemappedAssemblies(appConfigRemappedAssemblies, autoUnifiedRemappedAssemblies);
 
-                    // Compute all dependencies.
-                    dependencyTable.ComputeClosure(allRemappedAssemblies, _assemblyFiles, _assemblyNames, generalResolutionExceptions);
+                    IReadOnlyCollection<DependentAssembly> allRemappedAssemblies = CombineRemappedAssemblies(appConfigRemappedAssemblies, autoUnifiedRemappedAssemblies);
+                    List<DependentAssembly> idealAssemblyRemappings = null;
+                    List<AssemblyNameReference> idealAssemblyRemappingsIdentities = null;
+                    bool hasConflicts = autoUnifiedRemappedAssemblies != null && autoUnifiedRemappedAssemblies.Count > 0;
 
-                    try
+                    if (AutoUnify && FindDependencies && !hasConflicts && !excludedReferencesExist)
```

Comment:
There was a test failing since the first ```RemoveReferencesMarkedForExclusion()``` doesn't warn when references are removed and the second does, so if the second execution is skipped then no warning is given. So it's either force it to run again and warn, or also warn on the first execution but have duplicate warnings show up. Neither are really ideal.

Reply:
Or maybe warn the first time and don't warn the second time if it's a rerun?

Reply:
That was my first thought, but can the list of warnings change the second time through?

Reply:
That was my concern too, not 100% sure